sktime/clustering/spatio_temporal/_st_dbscan.py

```diff
@@ -0,0 +1,296 @@
+            raise ValueError(
+                'frame_size, frame_overlap not correctly configured.')
+
+        time = np.unique(X[:, 0])
+
+        labels = None
+        right_overlap = 0
+        max_label = 0
+
+        for i in range(0, len(time), (frame_size - frame_overlap + 1)):
+            for period in [time[i:i + frame_size]]:
+                frame = X[np.isin(X[:, 0], period)]
+
+                self.fit(frame)
+
+                if not type(labels) is np.ndarray:
+                    labels = self.labels
+                else:
+                    frame_one_overlap_labels = labels[len(labels) -
+                                                      right_overlap:]
+                    frame_two_overlap_labels = self.labels[0:right_overlap]
+
+                    mapper = {}
+                    for i in list(
+                            zip(frame_one_overlap_labels,
+                                frame_two_overlap_labels)):
+                        mapper[i[1]] = i[0]
+                    mapper[
+                        -1] = -1 
+
+                    ignore_clusters = set(
+                        self.labels) - set(frame_two_overlap_labels)
+                    if -1 in labels:
+                        labels_counter = len(set(labels)) - 1
+                    else:
+                        labels_counter = len(set(labels))
+                    for j in ignore_clusters:
+                        mapper[j] = labels_counter
+                        labels_counter += 1
+
+                    new_labels = np.array([mapper[j] for j in self.labels])
+
+                    labels = labels[0:len(labels) - right_overlap]
+                    labels = np.concatenate((labels, new_labels))
+
+                right_overlap = len(X[np.isin(X[:, 0],
+                                              period[-frame_overlap + 1:])])
+
+        self.labels = labels
+        return self
+
+    def _predict(self, X):
+        """Predict the closest cluster each sample in X belongs to.
+
+        Parameters
+        ----------
+        X : data to cluster based on model formed in _fit, of type self.get_tag(
+        "X_inner_mtype")
+        y: ignored, exists for API consistency reasons.
+
+        Returns
+        -------
+        np.ndarray (1d array of shape (n_instances,))
+            Index of the cluster each time series in X belongs to.
+        """
+
+        X = check_array(X)
+
+        dist_matrix = squareform(pdist(X, metric=self.dist))
+        predicted_labels = self.model_.fit_predict(dist_matrix)
+
+        return predicted_labels
+
+    def _get_fitted_params(self):
+        """Get fitted parameters.
+
+        private _get_fitted_params, called from get_fitted_params
+
+        State required:
+            Requires state to be "fitted".
+
+        Returns
+        -------
+        fitted_params : dict with str keys
+            fitted parameters, keyed by names of fitted parameter
+        """
+
+        fitted_params = {
```

Comment:
not all of these are fitted parameters, some are hyper-parameters from the init.

I would suggest, just delete `_get_fitted_params`, it automatically exposes any attribute ending in underscord.